packages/core/src/core.ts

```diff
@@ -91,6 +91,28 @@ export function getInput(name: string, options?: InputOptions): string {
   return val.trim()
 }
 
+/**
+ * Gets the input value of the boolean type in the YAML 1.2 "core schema" specification.
+ * Support boolean input list: `true | True | TRUE | false | False | FALSE` .
+ * The return value is also in boolean type.
+ * ref: https://yaml.org/spec/1.2/spec.html#id2804923
+ *
+ * @param     name     name of the input to get
+ * @param     options  optional. See InputOptions.
+ * @returns   boolean
+ */
+export function getBooleanInput(name: string, options?: InputOptions): boolean {
+  const trueValue = ['true', 'True', 'TRUE']
+  const falseValue = ['false', 'False', 'FALSE']
+  const val = getInput(name, options)
+  if (trueValue.includes(val)) return true
+  if (falseValue.includes(val)) return false
+  throw new TypeError(
```

Comment:
what if it was `required: false`?

Reply:
@chamini2 The `required: false` will deal with the `getInput`(Line 107) and get the default value. 👀

Reply:
But if there is **no default value**, that will make `val` an empty string, which we don't consider in line 108 or 109. Resulting in `throw new TypeError` because `trueValue.includes('')` and `falseValue.includes('')` are both false.

Reply:
> Action inputs can be read with `getInput` which returns a `string` or `getBooleanInput` which parses a boolean based on the [yaml 1.2 specification](https://yaml.org/spec/1.2/spec.html#id2804923). If required set to be false, the input should have a default value in `action.yml`.

https://github.com/yi-Xu-0100/toolkit/blob/main/packages/core/README.md

@chamini2 Why a boolean type input have no default value? 👀

Reply:
Oh, I did not know this was expected. If `required: false` and no default I thought it had just to return `null` or `''`. Thanks, sorry for the inconvenience.